### PR TITLE
Divert naming of rewards field in different epoch

### PR DESCRIPTION
**Pull request by kaikawaliu**
## Description

Since the `rewards` field in the actual TxBlock can mean for Transaction fees incurred during non-vacuous epoch and also coinbase reward during vacuous epoch. For avoid misunderstanding we decided to explain it into two fields in API level, `TxnFees` and `Rewards` respectively as below:
```
                  | TxnFees | Rewards
Non-vacuous       | fees    | 0
Vacuous           | 0       | coinbase
```
## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

**Review by evesnow91 — APPROVED** [review]

**Comment by KaustubhShamshery** [comment]
I don't think you should use the current block num instead we should use the block num of the tx block to figure out if it is a vacuous epoch.

**Comment by kaikawaliu** [comment]
> I don't think you should use the current block num instead we should use the block num of the tx block to figure out if it is a vacuous epoch.

Yes I'm doing this exactly. which line doesn't look correct to you?

**Review by ansnunez — APPROVED** [review]

File: src/libServer/JSONConversion.cpp [hunk 1]
```diff
@@ -58,7 +59,8 @@ const Json::Value JSONConversion::convertTxBlocktoJson(const TxBlock& txblock) {
   ret_head["Version"] = txheader.GetVersion();
   ret_head["GasLimit"] = to_string(txheader.GetGasLimit());
   ret_head["GasUsed"] = to_string(txheader.GetGasUsed());
-  ret_head["Rewards"] = txheader.GetRewards().str();
+  ret_head["Rewards"] = isVacuous ? txheader.GetRewards().str() : 0;
```
Comment (KaustubhShamshery) [inline src/libServer/JSONConversion.cpp:62]: here `isVacous` is governed by some outside parameter, why dont just call m_mediator.GetIsVacuousEpoch(Latest.GetHeader().GetBlockNum()) inside the function. This extra parameter is not needed

Comment (ansnunez) [reply src/libServer/JSONConversion.cpp:62]: Good point. Can use `txheader.GetBlockNum()`

**Review by KaustubhShamshery — APPROVED** [review]

**Pull request merged**